pkg/instrumentation/annotationmutator.go

```diff
@@ -88,18 +81,24 @@ func NewAnnotationMutator(mutations []AnnotationMutation) AnnotationMutator {
 	return AnnotationMutator{mutations: mutations}
 }
 
-// Mutate modifies the object's annotations based on the mutator's mutations. Returns whether any of the
-// mutations changed the annotations.
-func (m *AnnotationMutator) Mutate(obj metav1.Object) bool {
+// Mutate modifies the object's annotations based on the mutator's mutations. Returns all the mutated annotations and
+// whether any of the mutations changed the annotations.
+func (m *AnnotationMutator) Mutate(obj metav1.Object) (map[string]string, bool) {
 	annotations := obj.GetAnnotations()
 	if annotations == nil {
 		annotations = make(map[string]string)
 	}
 	var anyMutated bool
+	allMutatedAnnotations := make(map[string]string)
 	for _, mutation := range m.mutations {
-		mutated := mutation.Mutate(annotations)
+		mutatedAnnotations, mutated := mutation.Mutate(annotations)
 		anyMutated = anyMutated || mutated
+		if mutated {
```

Comment:
nit: Don't need the check. An empty or even nil map will just not do anything with a for-range loop.